src/main/resources/controllers-context.xml

```diff
@@ -24,19 +24,25 @@
         <constructor-arg ref="jdbcConnectionHolder"/>
     </bean>
 
-    <bean id="transactionalDogService" class="com.example.dog.service.TransactionalDogService">
-        <constructor-arg>
-            <bean class="com.example.dog.service.JdbcDogService">
-                <constructor-arg ref="jdbcDogDao"/>
-            </bean>
-        </constructor-arg>
-        <constructor-arg ref="jdbcConnectionHolder"/>
+    <bean id="jdbcDogService" class="com.example.dog.service.JdbcDogService" primary="true">
+        <constructor-arg ref="jdbcDogDao"/>
     </bean>
 
     <bean id="jdbcConnectionHolder" class="com.example.dog.dao.JdbcConnectionHolder">
         <constructor-arg ref="dataSource"/>
     </bean>
 
+    <bean id="transactionalDogService" class="com.example.dog.service.TransactionalDogService">
+        <constructor-arg ref="jdbcDogService"/>
+        <constructor-arg ref="jdbcConnectionHolder"/>
+    </bean>
+
+    <bean id="transactionalProxy" class="com.example.dog.service.TransactionalProxy" factory-method="createProxy">
```

Comment:
They all are proxies, even the one that you created manually. This one is a jdkDynamicProxy.

My advice is to create spring profiles and depending on the active profile - use this or that implementation of the proxy.

Reply:
So does this mean that I need to create another .xml file for proxy beans or I can do it using the existing configuration file?

Reply:
Same XML file. Look up Spring Profiles in XML.

Reply:
Please look at the next commit -> I have 2 profiles, one of them is active --> so, I expect that other beans (w/o specified profiles) will be created automatically, but it seems like Controller cannot see transactionalProxy bean ..?

Reply:
It doesn't see it in tests or when you start tomcat? I see that you activated the profile only for tomcat.

Reply:
I thought it should work for both types of tests